sklearn/utils/validation.py

```diff
@@ -1076,7 +1076,11 @@ def check_X_y(
         performed if the dtype of the input is not in the list.
 
     order : {'F', 'C'}, default=None
-        Whether an array will be forced to be fortran or c-style.
+        Whether an array will be forced to be fortran or c-style. If
+        `order=None` and `copy=True`, the output array memory layout is kept
+        similar to the original (using the `np.array` function). If
+        `order=None` and `copy=False`, the memory layout of the output array is
+        not guaranteed (using the `np.asarray` function).
```

Comment:
I think this is too much information about the implementation details. Here is my suggestion:

```suggestion
        Whether an array will be forced to be fortran or c-style. If
        `None`, then the input data's order is preserved when possible.
```

Here is how I came to this conclusion about the behavior:

- For `copy=True`, we use `np.array`, which [preserves order when possible](https://numpy.org/doc/stable/reference/generated/numpy.array.html).
- For `copy=False`, we use `np.asarray`, which [preserves order by default](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html).

Note in both cases, passing in `order=None` means `order="K"` in their documentation.

Reply:
Sounds good, thanks for the review and suggestion!